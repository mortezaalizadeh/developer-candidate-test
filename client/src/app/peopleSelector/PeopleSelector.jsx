import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { withTranslation } from 'react-i18next';
import Styles from './Styles';

const PeopleSelector = ({ classes, t, selectedFilter, onSelectorSelected }) => {
  return (
    <Paper className={classes.root}>
      <Typography>
        <Button onClick={() => onSelectorSelected('everybody')} color={selectedFilter === 'everybody' ? 'primary' : 'default'}>
          {t('everybody.label')}
        </Button>
        <Button onClick={() => onSelectorSelected('male')} color={selectedFilter === 'male' ? 'primary' : 'default'}>
          {t('male.label')}
        </Button>
        <Button onClick={() => onSelectorSelected('female')} color={selectedFilter === 'female' ? 'primary' : 'default'}>
          {t('female.label')}
        </Button>
        <Button onClick={() => onSelectorSelected('over30')} color={selectedFilter === 'over30' ? 'primary' : 'default'}>
          {t('over30.label')}
        </Button>
        <Button onClick={() => onSelectorSelected('under30')} color={selectedFilter === 'under30' ? 'primary' : 'default'}>
          {t('under30.label')}
        </Button>
      </Typography>
    </Paper>
  );
};

PeopleSelector.propTypes = {
  classes: PropTypes.object.isRequired,
  t: PropTypes.func.isRequired,
  selectedFilter: PropTypes.string.isRequired,
  onSelectorSelected: PropTypes.func.isRequired,
};

export default withStyles(Styles)(withTranslation()(PeopleSelector));
