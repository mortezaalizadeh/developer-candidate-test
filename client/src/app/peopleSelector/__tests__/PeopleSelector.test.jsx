import React from 'react';
import renderer from 'react-test-renderer';
import PeopleSelector from '../PeopleSelector';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<PeopleSelector />).toJSON();

  expect(tree).toMatchSnapshot();
});
