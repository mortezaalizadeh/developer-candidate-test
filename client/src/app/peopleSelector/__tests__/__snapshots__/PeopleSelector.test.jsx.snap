// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5 Component-root-1"
>
  <p
    className="MuiTypography-root-31 MuiTypography-body1-40"
  >
    <button
      className="MuiButtonBase-root-93 MuiButton-root-67 MuiButton-text-69 MuiButton-flat-72"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-68"
      >
        everybody.label
      </span>
      <span
        className="MuiTouchRipple-root-96"
      />
    </button>
    <button
      className="MuiButtonBase-root-93 MuiButton-root-67 MuiButton-text-69 MuiButton-flat-72"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-68"
      >
        male.label
      </span>
      <span
        className="MuiTouchRipple-root-96"
      />
    </button>
    <button
      className="MuiButtonBase-root-93 MuiButton-root-67 MuiButton-text-69 MuiButton-flat-72"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-68"
      >
        female.label
      </span>
      <span
        className="MuiTouchRipple-root-96"
      />
    </button>
    <button
      className="MuiButtonBase-root-93 MuiButton-root-67 MuiButton-text-69 MuiButton-flat-72"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-68"
      >
        over30.label
      </span>
      <span
        className="MuiTouchRipple-root-96"
      />
    </button>
    <button
      className="MuiButtonBase-root-93 MuiButton-root-67 MuiButton-text-69 MuiButton-flat-72"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-68"
      >
        under30.label
      </span>
      <span
        className="MuiTouchRipple-root-96"
      />
    </button>
  </p>
</div>
`;
